presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcDistributedQueries.java

```diff
@@ -13,21 +13,67 @@
  */
 package io.prestosql.plugin.jdbc;
 
+import com.google.common.collect.ImmutableList;
 import io.airlift.tpch.TpchTable;
 import io.prestosql.tests.AbstractTestQueries;
+import io.prestosql.tests.sql.JdbcSqlExecutor;
+import io.prestosql.tests.sql.TestTable;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
 
 import static io.prestosql.plugin.jdbc.JdbcQueryRunner.createJdbcQueryRunner;
+import static io.prestosql.tests.sql.TestTable.TABLE_NAME_PLACEHOLDER;
+import static java.lang.String.format;
 
 public class TestJdbcDistributedQueries
         extends AbstractTestQueries
 {
+    private final Map<String, String> properties;
+
     public TestJdbcDistributedQueries()
     {
-        super(() -> createJdbcQueryRunner(TpchTable.getTables()));
+        this(TestingH2JdbcModule.createProperties());
+    }
+
+    public TestJdbcDistributedQueries(Map<String, String> properties)
+    {
+        super(() -> createJdbcQueryRunner(ImmutableList.copyOf(TpchTable.getTables()), properties));
+        this.properties = properties;
     }
 
     @Override
     public void testLargeIn()
     {
     }
+
+    @Test
+    public void testUnknownType()
+    {
+        try (TestTable table = new TestTable(
+                getSqlExecutor(),
+                "tpch.test_failure_on_unknown_type",
+                format("CREATE TABLE %s (i int, x GEOMETRY)", TABLE_NAME_PLACEHOLDER),
+                Optional.of("(1, 'POINT(7 52)')"))) {
+            assertQuery("SELECT i FROM " + table.getName(), "VALUES 1");
+            assertQueryFails("SELECT * FROM " + table.getName(), "Unsupported column type for: X");
+            assertQuery(
+                    "SELECT column_name, data_type FROM information_schema.columns WHERE table_name LIKE '%_unknown_%'",
+                    "VALUES ('i', 'integer'), ('x', 'unknown')");
+            assertQuery("DESCRIBE " + table.getName(), "VALUES ('i', 'integer', '', ''), ('x', 'unknown', '', '')");
+
+            assertQueryFails(
+                    format("INSERT INTO %s VALUES (1, 1)", table.getName()),
+                    "\\QInsert query has mismatched column types: Table: [integer, unknown], Query: [integer, integer]\\E");
+            assertQueryFails(format("INSERT INTO %s VALUES (1, null)", table.getName()), "Unsupported column type: unknown");
+            assertQueryFails(format("INSERT INTO %s (i) VALUES (1)", table.getName()), "UNKNOWN is not a valid type");
```

Comment:
This should succeed. https://github.com/prestosql/presto/issues/1185
Add a TODO